Kconfig.zephyr

```diff
@@ -350,7 +350,7 @@ config MAKEFILE_EXPORTS
 
 config LEGACY_DEVICETREE_MACROS
 	bool "Allow use of legacy devicetree macros"
-	default y
+	default n
```

Comment:
Can just remove the line. This will probably anger the linter.

Reply:
Doesn't this break mcuboot?

```
=== running "rg DT_FLASH_AREA" in mcuboot (bootloader/mcuboot):
scripts/assemble.py
37:offset_re = re.compile(r"^#define DT_FLASH_AREA_([0-9A-Z_]+)_OFFSET(_0)?\s+(0x[0-9a-fA-F]+|[0-9]+)$")
38:size_re   = re.compile(r"^#define DT_FLASH_AREA_([0-9A-Z_]+)_SIZE(_0)?\s+(0x[0-9a-fA-F]+|[0-9]+)$")
boot/zephyr/include/target.h
38:    !defined(DT_FLASH_AREA_IMAGE_0_OFFSET) || \
39:    !defined(DT_FLASH_AREA_IMAGE_0_SIZE) || \
40:    !defined(DT_FLASH_AREA_IMAGE_1_OFFSET) || \
41:    !defined(DT_FLASH_AREA_IMAGE_1_SIZE) || \
42:    (!defined(CONFIG_BOOT_SWAP_USING_MOVE) && !defined(DT_FLASH_AREA_IMAGE_SCRATCH_OFFSET)) || \
43:    (!defined(CONFIG_BOOT_SWAP_USING_MOVE) && !defined(DT_FLASH_AREA_IMAGE_SCRATCH_SIZE))
sim/mcuboot-sys/csupport/devicetree.h
12:#define DT_FLASH_AREA_IMAGE_0_ID 1
13:#define DT_FLASH_AREA_IMAGE_1_ID 2
14:#define DT_FLASH_AREA_IMAGE_SCRATCH_ID 3
15:#define DT_FLASH_AREA_IMAGE_2_ID 4
16:#define DT_FLASH_AREA_IMAGE_3_ID 5
boot/zephyr/include/sysflash/sysflash.h
16:                                         DT_FLASH_AREA_IMAGE_0_ID : \
17:                                         DT_FLASH_AREA_IMAGE_0_ID)
19:                                         DT_FLASH_AREA_IMAGE_1_ID : \
20:                                         DT_FLASH_AREA_IMAGE_1_ID)
27:                                         DT_FLASH_AREA_IMAGE_0_ID : \
29:                                         DT_FLASH_AREA_IMAGE_2_ID : \
32:                                         DT_FLASH_AREA_IMAGE_1_ID : \
34:                                         DT_FLASH_AREA_IMAGE_3_ID : \
41:#define FLASH_AREA_IMAGE_SCRATCH    DT_FLASH_AREA_IMAGE_SCRATCH_ID
```

Reply:
Fixed mcuboot.